evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -801,3 +801,38 @@ def test_component_graph_dataset_with_target_imputer():
     component_graph.fit(X, y)
     predictions = component_graph.predict(X)
     assert not pd.isnull(predictions.to_series()).any()
+
+
+@patch('evalml.pipelines.components.estimators.LogisticRegressionClassifier.fit')
+def test_component_graph_sampler_y_passes(mock_estimator_fit):
+    pytest.importorskip("imblearn.over_sampling", reason="Cannot import imblearn, skipping tests")
+    # makes sure the y value from oversampler gets passed to the estimator, even though StandardScaler has no y output
+    X = pd.DataFrame({"a": [i for i in range(100)],
+                      "b": [i % 3 for i in range(100)]})
+    y = pd.Series([0] * 90 + [1] * 10)
+    component_list = ['Imputer', 'SMOTE Oversampler', 'Standard Scaler', 'Logistic Regression Classifier']
+    component_graph = ComponentGraph.from_list(component_list)
+    component_graph.instantiate({})
+    component_graph.fit(X, y)
+    assert len(mock_estimator_fit.call_args[0][0]) == len(mock_estimator_fit.call_args[0][1])
+    assert len(mock_estimator_fit.call_args[0][0]) == int(1.25 * 90)
+
+
+@patch('evalml.pipelines.components.estimators.RandomForestClassifier.fit')
+@patch('evalml.pipelines.components.estimators.DecisionTreeClassifier.fit')
+def test_component_graph_sampler_same_given_components(mock_dt_fit, mock_rf_fit):
+    pytest.importorskip("imblearn.over_sampling", reason="Cannot import imblearn, skipping tests")
+    X = pd.DataFrame({"a": [i for i in range(100)],
+                      "b": [i % 3 for i in range(100)]})
+    y = pd.Series([0] * 90 + [1] * 10)
+    component_list = ['Imputer', 'SMOTE Oversampler', 'Random Forest Classifier']
```

Comment:
How come we need both `component_graph.fit` and `component_graph2.fit` if we're only checking the output of the mocks after `component_graph2.fit`? I think we need a `reset_mock` in between if we want to keep both.

Reply:
These are two different mocks. One is mocking the RFC pipeline, and the other mocks the DTC pipeline